docs/fundamentals/clean-code.md

```diff
@@ -263,4 +263,64 @@ with f:
 ```
 ````
 
+## Power heavy-lifting job by the Flow, not Client
+   
+Wrap heavy lifting jobs inside an `Executor` and part of the `Flow`.
+For instance, loading high resolution images from Client into `Document` and `DocumentArray`
+could be time consuming.
+Put it inside an Executor could leverage Jina's capability to scale it up.
```

Comment:
I view that this best practice mainly helps to reduce data sent over the network, if access to the uri is the same for the server and the client. Also this helps perform a lazy conversion, meaning it's converted on the fly.
I don't understand how Executor and Flow help with scaling up, unless the server's machine is faster than the client

Reply:
your comment make sense, I see it helps because scaling up a flow, e.g. using `parallel=n` or scale flow with k8s is easier. Not because server is faster than client.